FrontEnd/src/HomePage/MainPage.jsx

```diff
@@ -30,12 +39,25 @@ function MainPage({ userUID }) {
           <p>No posts found.</p>
         ) : (
           userFeed.map((post, index) => (
-            <div key={index}>
-              <p>{post.serviceCategory}</p>
-              <img src={post.imageURL} alt={`Post ${index}`} />
+            <div
+              key={index}
+              className="postFeed"
+              onClick={() => toggleModal(post)}
+            >
+              <PostsPreview post={post} />
             </div>
           ))
         )}
+        {isShowModal && selectedPost && (
```

Comment:
Explicit null check here?
```suggestion
        {isShowModal && selectedPost != null && (
```